nox/sessions.py

```diff
@@ -488,6 +489,15 @@ def install(self, *args: str, **kwargs: Any) -> None:
             raise ValueError(
                 "A session without a virtualenv can not install dependencies."
             )
+        if isinstance(venv, PassthroughEnv):
+            warnings.warn(
+                f"""Session {self.name} does not have a virtual environment, """
+                """so use of session.install() is deprecated since it would modify """
+                """the global Python environment. If you're really sure that is """
+                """what you want to do, use session.run("pip", "install", ...) instead.""",
+                category=DeprecationWarning,
+                stacklevel=2,
+            )
```

Comment:
This needs to be `FutureWarning` for it to be displayed to end users.

I also took the liberty to replace the triple quotes, and surround the session name with single quotes. Feel free to ignore.

```suggestion
            warnings.warn(
                f"Session {self.name!r} does not have a virtual environment, "
                "so use of session.install() is deprecated since it would modify "
                "the global Python environment. If you're really sure that is "
                'what you want to do, use session.run("pip", "install", ...) instead.',
                category=FutureWarning,
                stacklevel=2,
            )
```

FTR: The `stacklevel` setting ensures that users see the file name and line number of their noxfile, instead of a location in the Nox source code. 👍

Reply:
Thanks for the review!